libcaf_core/src/uri.cpp

```diff
@@ -101,6 +102,27 @@ std::string to_string(const uri& x) {
   return result;
 }
 
+std::string to_string(const uri::authority_type& x) {
+  std::string str;
+  if (!x.userinfo.empty()) {
+    detail::append_uri(str, x.userinfo);
+    str += '@';
+  }
+  auto addr = get_if<ip_address>(&x.host);
+  if (addr == nullptr) {
```

Comment:
```suggestion
  if (auto addr = get_if<ip_address>(&x.host)) {
```

(note: switches the `if` and `else` branch)